### PR TITLE
Merge v 8.2 into gsl/develop

### PR DESCRIPTION
**Pull request by jderrico-noaa**
Merges v 8.2 into gsl/develop

There are a number of conflicts present that begin with "CURRENT GSL/DEVELOP VERSION" that need to be manually resolved before the model can actually be compiled/tested

### CODE REVIEW

File: src/core_atmosphere/physics/mpas_atmphys_driver.F [hunk 1]
```diff
@@ -260,23 +268,23 @@ subroutine physics_driver(domain,itimestep,xtime_s)
...
 !$OMP END PARALLEL DO
        call deallocate_sfclayer(block%configs)
     endif
 
     !call to 1d ocean mixed-layer model
-    if(config_oml1d) call driver_oml1d(block%configs, mesh, diag, diag_physics, sfc_input)
+    if(config_oml1d) call driver_oml1d(block%configs,mesh,diag,diag_physics,sfc_input)
 
     !call to land-surface scheme:
     if(config_lsm_scheme .ne. 'off') then
```
Comment (jderrico-noaa) [inline src/core_atmosphere/physics/mpas_atmphys_driver.F]: @tanyasmirnova this is the if statement you said need to change

File: src/core_atmosphere/Registry.xml [hunk 2]
```diff
@@ -1607,34 +1704,35 @@
...
                              packages="mp_nssl2m_in"/>
 
                         <var name="ni" array_group="number" units="nb kg^{-1}"
                              description="Cloud ice number concentration"
-                             packages="bl_mynn_in;mp_thompson_in;mp_nssl2m_in"/>
+                             packages="bl_mynn_in;mp_thompson_in;mp_thompson_aers_in;mp_nssl2m_in"/>
 
                         <var name="nr" array_group="number" units="nb kg^{-1}"
                              description="Rain number concentration"
+     <!--CURRENT GSL/DEVELOP VERSION-->
```
Comment (jderrico-noaa) [inline src/core_atmosphere/Registry.xml]: @AndersJensen-NOAA

File: src/core_atmosphere/Registry.xml [hunk 3]
```diff
@@ -1983,45 +2105,46 @@
...
-                             packages="bl_mynn_in;mp_thompson_in;mp_nssl2m_in"/>
+                             packages="bl_mynn_in;mp_thompson_in;mp_thompson_aers_in;mp_nssl2m_in"/>
 
 			<var name="tend_nc" name_in_code="nc" array_group="number" units="nb m^{-3} s^{-1}" 
                              description="Tendency of cloud number concentration multiplied by dry air density divided by d(zeta)/dz"
                              packages="mp_thompson_in;mp_nssl2m_in"/>
 			
                         <var name="tend_nr" name_in_code="nr" array_group="number" units="nb m^{-3} s^{-1}" 
                              description="Tendency of rain number concentration multiplied by dry air density divided by d(zeta)/dz"
+     <!--CURRENT GSL/DEVELOP VERSION-->
```
Comment (jderrico-noaa) [inline src/core_atmosphere/Registry.xml]: @AndersJensen-NOAA

File: src/core_atmosphere/Registry.xml [hunk 4]
```diff
@@ -2095,33 +2234,42 @@
...
+                        <var name="lbc_nr" name_in_code="nr" array_group="number" packages="mp_thompson_in;mp_nssl2m_in;mp_thompson_aers_in"
+                             units="m^{-3} s^{-1}"
                              description="Lateral boundary tendency of rain number concentration"/>
 
-                        <var name="lbc_ni" name_in_code="ni" array_group="number" packages="bl_mynn_in;mp_thompson_in;mp_nssl2m_in" units="m^{-3} s^{-1}"
+                        <var name="lbc_ni" name_in_code="ni" array_group="number" packages="bl_mynn_in;mp_thompson_in;mp_thompson_aers_in;mp_nssl2m_in"
+                             units="m^{-3} s^{-1}"
                              description="Lateral boundary tendency of ice number concentration"/>
 
+     <!--CURRENT GSL/DEVELOP VERSION-->
```
Comment (jderrico-noaa) [inline src/core_atmosphere/Registry.xml]: @AndersJensen-NOAA

File: src/core_atmosphere/physics/mpas_atmphys_interface.F [hunk 5]
```diff
@@ -483,7 +545,11 @@ subroutine microphysics_from_MPAS(configs,mesh,state,time_lev,diag,diag_physics,
  integer,intent(in):: its,ite
  integer:: time_lev
 
+!inout variables:
+ type(mpas_pool_type),intent(inout):: tend_physics
+
 !local pointers:
+! CURRENT GSL/DEVELOP VERSION
```
Comment (jderrico-noaa) [inline src/core_atmosphere/physics/mpas_atmphys_interface.F]: @AndersJensen-NOAA

File: src/core_atmosphere/physics/mpas_atmphys_interface.F [hunk 6]
```diff
@@ -501,17 +573,23 @@ subroutine microphysics_from_MPAS(configs,mesh,state,time_lev,diag,diag_physics,
...
+ real(kind=RKIND),dimension(:,:),pointer  :: rthmpten,rqvmpten,rqcmpten,rqrmpten,rqimpten,rqsmpten,rqgmpten
+ real(kind=RKIND),dimension(:,:),pointer  :: rncmpten,rnimpten,rnrmpten,rnifampten,rnwfampten
  real(kind=RKIND),dimension(:,:,:),pointer:: scalars
 
 !local variables:
  integer:: i,k,j
 
 !-----------------------------------------------------------------------------------------------------------------
 
+! CURRENT GSL/DEVELOP VERSION
```
Comment (jderrico-noaa) [inline src/core_atmosphere/physics/mpas_atmphys_interface.F]: @AndersJensen-NOAA

File: src/core_atmosphere/physics/mpas_atmphys_interface.F [hunk 7]
```diff
@@ -584,6 +662,7 @@ subroutine microphysics_from_MPAS(configs,mesh,state,time_lev,diag,diag_physics,
  enddo
  enddo
 
+! CURRENT GSL/DEVELOP VERSION
```
Comment (jderrico-noaa) [inline src/core_atmosphere/physics/mpas_atmphys_interface.F]: @AndersJensen-NOAA

File: src/core_atmosphere/physics/mpas_atmphys_interface.F [hunk 8]
```diff
@@ -592,20 +671,42 @@ subroutine microphysics_from_MPAS(configs,mesh,state,time_lev,diag,diag_physics,
...
+          evapprod_p(i,k,j) = evapprod(k,k)
+          recloud_p(i,k,j)  = re_cloud(k,i)
+          reice_p(i,k,j)    = re_ice(k,i)
+          resnow_p(i,k,j)   = re_snow(k,i)
           refl10cm_p(i,k,j) = refl10cm(k,i)
        enddo
        enddo
        enddo
 
+! CURRENT GSL/DEVELOP VERSION
```
Comment (jderrico-noaa) [inline src/core_atmosphere/physics/mpas_atmphys_interface.F]: @AndersJensen-NOAA

File: src/core_atmosphere/physics/mpas_atmphys_interface.F [hunk 9]
```diff
@@ -691,43 +793,159 @@ subroutine microphysics_from_MPAS(configs,mesh,state,time_lev,diag,diag_physics,
...
+!inout variables:
  type(mpas_pool_type),intent(inout):: state
  type(mpas_pool_type),intent(inout):: diag
  type(mpas_pool_type),intent(inout):: tend
  type(mpas_pool_type),intent(inout):: diag_physics
+ type(mpas_pool_type),intent(inout):: tend_physics
+
 
 !local pointers:
+ ! CURRENT GSL/DEVELOP VERSION
```
Comment (jderrico-noaa) [inline src/core_atmosphere/physics/mpas_atmphys_interface.F]: @AndersJensen-NOAA

File: src/core_atmosphere/physics/mpas_atmphys_interface.F [hunk 10]
```diff
@@ -747,9 +967,13 @@ subroutine microphysics_to_MPAS(configs,mesh,state,time_lev,diag,diag_physics,te
 
 !-----------------------------------------------------------------------------------------------------------------
 
+! CURRENT GSL/DEVELOP VERSION
```
Comment (jderrico-noaa) [inline src/core_atmosphere/physics/mpas_atmphys_interface.F]: @AndersJensen-NOAA

File: src/core_atmosphere/physics/mpas_atmphys_interface.F [hunk 11]
```diff
@@ -857,20 +1081,41 @@ subroutine microphysics_to_MPAS(configs,mesh,state,time_lev,diag,diag_physics,te
  enddo
  enddo
 
+! CURRENT GSL/DEVELOP VERSION
```
Comment (jderrico-noaa) [inline src/core_atmosphere/physics/mpas_atmphys_interface.F]: @AndersJensen-NOAA

File: src/core_atmosphere/physics/mpas_atmphys_interface.F [hunk 12]
```diff
@@ -879,6 +1124,7 @@ subroutine microphysics_to_MPAS(configs,mesh,state,time_lev,diag,diag_physics,te
        enddo
        enddo
 
+! CURRENT GLS/DEVELOP VERSION
```
Comment (jderrico-noaa) [inline src/core_atmosphere/physics/mpas_atmphys_interface.F]: @AndersJensen-NOAA

File: src/core_atmosphere/physics/mpas_atmphys_interface.F [hunk 13]
```diff
@@ -922,7 +1168,16 @@ subroutine microphysics_to_MPAS(configs,mesh,state,time_lev,diag,diag_physics,te
...
+! INCOMING 8.2 VERSION       
+mp2_select: select case(trim(mp_scheme))
+          case("mp_thompson","mp_thompson_aerosols")
+             call mpas_pool_get_dimension(state,'index_ni',index_ni)
+             call mpas_pool_get_dimension(state,'index_nr',index_nr)
+             ni   => scalars(index_ni,:,:)
+             nr   => scalars(index_nr,:,:)
+! MODEL CANNOT BE COMPILED UNTIL THIS IS RESOLVED - JMD
+
+! CURRENT GSL/DEVELOP VERSION
```
Comment (jderrico-noaa) [inline src/core_atmosphere/physics/mpas_atmphys_interface.F]: @AndersJensen-NOAA

File: src/core_atmosphere/physics/mpas_atmphys_driver_microphysics.F [hunk 14]
```diff
@@ -316,31 +332,35 @@ subroutine deallocate_microphysics(configs)
...
     end select microp2_select
 
     case default
-
  end select microp_select
 
  end subroutine deallocate_microphysics
 
 !=================================================================================================================
+!CURRENT GSL/DEVELOP VERSION
```
Comment (jderrico-noaa) [inline src/core_atmosphere/physics/mpas_atmphys_driver_microphysics.F]: @AndersJensen-NOAA

File: src/core_atmosphere/physics/mpas_atmphys_driver_microphysics.F [hunk 15]
```diff
@@ -352,16 +372,23 @@ subroutine microphysics_init(dminfo,configs,mesh,sfc_input,diag,diag_physics,sta
...
 !-----------------------------------------------------------------------------------------------------------------
+!call mpas_log_write(' ')
+!call mpas_log_write('--- enter subroutine init_microphysics:')
 
 !initialization of CCPP-compliant flags:
  errmsg = ' '
  errflg = 0
 
  call mpas_pool_get_config(configs,'config_microp_scheme',microp_scheme)
+! CURRENT GSL/DEVELOP VERSION
```
Comment (jderrico-noaa) [inline src/core_atmosphere/physics/mpas_atmphys_driver_microphysics.F]: @AndersJensen-NOAA

File: src/core_atmosphere/physics/mpas_atmphys_driver_microphysics.F [hunk 16]
```diff
@@ -352,16 +372,23 @@ subroutine microphysics_init(dminfo,configs,mesh,sfc_input,diag,diag_physics,sta
...
  call mpas_pool_get_config(configs,'config_thompson_aerosol_aware',thompson_aerosol_aware)
  call mpas_pool_get_config(configs,'config_thompson_hail_aware',thompson_hail_aware)
   call mpas_pool_get_config(configs,'config_nssl_moments',nssl_moments)
+! INCOMING 8.2 VERSION
+   call mpas_pool_get_config(configs,'config_do_restart'   ,do_restart   )
+! MODEL CANNOT BE COMPILED UNTIL THIS IS RESOLVED - JMD
 
+! CURRENT GSL/DEVELOP VERSION
```
Comment (jderrico-noaa) [inline src/core_atmosphere/physics/mpas_atmphys_driver_microphysics.F]: @AndersJensen-NOAA

File: src/core_init_atmosphere/mpas_init_atm_cases.F [hunk 17]
```diff
@@ -271,7 +272,11 @@ subroutine init_atm_setup_case(domain, stream_manager)
                                    diag, diag_physics, block_ptr % dimensions, block_ptr % configs)
 
             if (config_met_interp) then
+! CURRENT GSL/DEVELOP VERSION
```
Comment (jderrico-noaa) [inline src/core_init_atmosphere/mpas_init_atm_cases.F]: @AndersJensen-NOAA

File: src/core_init_atmosphere/mpas_init_atm_cases.F [hunk 18]
```diff
@@ -358,10 +363,14 @@ subroutine init_atm_setup_case(domain, stream_manager)
                call init_atm_case_lbc(timeString, block_ptr, mesh, nCells, nEdges, nVertLevels, fg, state, &
                                       diag, lbc_state, block_ptr % dimensions, block_ptr % configs)
 
+! CURRENT GSL/DEVELOP VERSION
```
Comment (jderrico-noaa) [inline src/core_init_atmosphere/mpas_init_atm_cases.F]: @AndersJensen-NOAA

File: src/core_init_atmosphere/Registry.xml [hunk 19]
```diff
@@ -1122,11 +1171,11 @@
 
                         <var name="qr" array_group="moist" units="kg kg^{-1}"
                              description="Rain water mixing ratio"/>
-
-			<var name="nwfa" array_group="passive" units="kg^{-1}"
+     <!--CURRENT GSL/DEVELOP VERSION-->
```
Comment (jderrico-noaa) [inline src/core_init_atmosphere/Registry.xml]: @AndersJensen-NOAA

File: src/core_init_atmosphere/Registry.xml [hunk 20]
```diff
@@ -1189,11 +1248,11 @@
 
                         <var name="lbc_qr" array_group="moist" units="kg kg^{-1}"
                              description="Rain water mixing ratio on lateral boundary cells"/>
-
+     <!--CURRENT GSL/DEVELOP VERSION-->
```
Comment (jderrico-noaa) [inline src/core_init_atmosphere/Registry.xml]: @AndersJensen-NOAA

File: src/core_atmosphere/physics/mpas_atmphys_driver.F (hunk 1 above)
Comment (tanyasmirnova) [reply src/core_atmosphere/physics/mpas_atmphys_driver.F]: The line 281 should be:
if(config_lsm_scheme .eq. 'sf_noah' or. config_lsm_scheme .eq. 'sf_ruc') then

**Review by tanyasmirnova — CHANGES_REQUESTED** [review]
@jderrico-noaa Janet, PLease, change line 281 in mpas_atmphys_driver.F to:
if(config_lsm_scheme .eq. 'sf_noah' or. config_lsm_scheme .eq. 'sf_ruc') then

File: src/core_atmosphere/physics/mpas_atmphys_driver.F (hunk 1 above)
Comment (jderrico-noaa) [inline src/core_atmosphere/physics/mpas_atmphys_driver.F]:
```suggestion
   if(config_lsm_scheme .eq. 'sf_noah' or. config_lsm_scheme .eq. 'sf_ruc') then
```

Comment (jderrico-noaa) [inline src/core_atmosphere/physics/mpas_atmphys_driver.F]:
```suggestion
    if(config_lsm_scheme .eq. 'sf_noah' .or. config_lsm_scheme .eq. 'sf_ruc') then
```

File: src/core_atmosphere/physics/mpas_atmphys_driver.F [hunk 21]
```diff
@@ -260,23 +268,23 @@ subroutine physics_driver(domain,itimestep,xtime_s)
...
        call deallocate_sfclayer(block%configs)
     endif
 
     !call to 1d ocean mixed-layer model
-    if(config_oml1d) call driver_oml1d(block%configs, mesh, diag, diag_physics, sfc_input)
+    if(config_oml1d) call driver_oml1d(block%configs,mesh,diag,diag_physics,sfc_input)
 
     !call to land-surface scheme:
-    if(config_lsm_scheme .ne. 'off') then
+   if(config_lsm_scheme .eq. 'sf_noah' or. config_lsm_scheme .eq. 'sf_ruc') then
```
Comment (jderrico-noaa) [inline src/core_atmosphere/physics/mpas_atmphys_driver.F]:
```suggestion
   if(config_lsm_scheme .eq. 'sf_noah' .or. config_lsm_scheme .eq. 'sf_ruc') then
```

File: src/core_atmosphere/physics/mpas_atmphys_driver_pbl.F [hunk 22]
```diff
@@ -10,6 +10,7 @@ module mpas_atmphys_driver_pbl
  use mpas_kind_types
  use mpas_pool_routines
  use mpas_timer,only: mpas_timer_start,mpas_timer_stop
+ use mpas_timer,only: mpas_timer_start,mpas_timer_stop
```
Comment (joeolson42) [inline src/core_atmosphere/physics/mpas_atmphys_driver_pbl.F]: duplicate line

File: src/core_atmosphere/physics/mpas_atmphys_driver_pbl.F [hunk 23]
```diff
@@ -24,6 +25,7 @@ module mpas_atmphys_driver_pbl
  public:: allocate_pbl,   &
           deallocate_pbl, &
           init_pbl,       &
+          init_pbl,       &
```
Comment (joeolson42) [inline src/core_atmosphere/physics/mpas_atmphys_driver_pbl.F]: duplicate line

File: src/core_atmosphere/physics/mpas_atmphys_driver_sfclayer.F [hunk 24]
```diff
@@ -15,10 +15,12 @@ module mpas_atmphys_driver_sfclayer
  use mpas_atmphys_constants
  use mpas_atmphys_vars
 
+ use module_sf_mynn,only: sfclay_mynn
```
Comment (joeolson42) [inline src/core_atmosphere/physics/mpas_atmphys_driver_sfclayer.F:18]: duplicate line

File: src/core_atmosphere/physics/mpas_atmphys_driver_sfclayer.F [hunk 25]
```diff
@@ -15,10 +15,12 @@ module mpas_atmphys_driver_sfclayer
...
  use module_sf_mynn,only: sfclay_mynn
  use module_sf_sfclay
  use module_sf_sfclayrev,only: sfclayrev
  use sf_mynn,only: sf_mynn_init
+ use sf_mynn,only: sf_mynn_init
```
Comment (joeolson42) [inline src/core_atmosphere/physics/mpas_atmphys_driver_sfclayer.F]: duplicate line

File: src/core_atmosphere/physics/mpas_atmphys_packages.F [hunk 26]
```diff
@@ -39,10 +39,12 @@ function atmphys_setup_packages(configs,streamInfo,packages,iocontext) result(ie
  character(len=StrKIND),pointer:: config_convection_scheme
  character(len=StrKIND),pointer:: config_pbl_scheme
  character(len=StrKIND),pointer:: config_lsm_scheme
- logical,pointer:: mp_kessler_in,mp_thompson_in,mp_wsm6_in,mp_nssl2m_in,nssl3m_in
- logical,pointer:: cu_grell_freitas_in,cu_kain_fritsch_in,cu_ntiedtke_in
+ character(len=StrKIND),pointer:: config_lsm_scheme
```
Comment (joeolson42) [inline src/core_atmosphere/physics/mpas_atmphys_packages.F]: duplicate line with config_lsm_scheme

File: src/core_atmosphere/physics/mpas_atmphys_packages.F [hunk 27]
```diff
@@ -39,10 +39,12 @@ function atmphys_setup_packages(configs,streamInfo,packages,iocontext) result(ie
...
+ logical,pointer:: mp_kessler_in,mp_thompson_in,mp_thompson_aers_in,mp_wsm6_in,mp_nssl2m_in,nssl3m_in
+ logical,pointer:: cu_grell_freitas_in,cu_kain_fritsch_in,cu_nntiedtke_in
```
Comment (joeolson42) [inline src/core_atmosphere/physics/mpas_atmphys_packages.F]: I think "cu_nntiedtke_in" should be "cu_ntiedtke_in"

File: src/core_atmosphere/physics/mpas_atmphys_vars.F [hunk 28]
```diff
@@ -109,6 +111,7 @@ module mpas_atmphys_vars
 !   Laura D. Fowler (laura@ucar.edu) / 2017-04-19.
 ! * added the local variables cosa_p and sina_p needed in call to subroutine gwdo after updating module_bl_gwdo.F
 !   to that of WRF version 4.0.2.
+!   to that of WRF version 4.0.2.
```
Comment (joeolson42) [inline src/core_atmosphere/physics/mpas_atmphys_vars.F]: inconsequential duplicate line

File: src/core_atmosphere/physics/mpas_atmphys_vars.F [hunk 29]
```diff
@@ -893,6 +925,12 @@ module mpas_atmphys_vars
     frc_urb_p,        &!urban fraction                                                                         [-]
     ust_urb_p          !urban u* in similarity theory                                                        [m/s]
 
+!=================================================================================================================
+!.. variables and arrays related to the Noahmp land-surface parameterization:
+!=================================================================================================================
+
+ type(NoahmpIO_type):: mpas_noahmp
+
 !-- arrays needed to run RUC LSM
```
Comment (joeolson42) [inline src/core_atmosphere/physics/mpas_atmphys_vars.F:940]: may want to addd the same style, i.e.:
!===========================================
before and after this line

File: src/core_atmosphere/physics/physics_wrf/module_sf_mynn.F [hunk 30]
```diff
@@ -391,6 +465,13 @@ subroutine sfclay_mynn(                                         &
 
  end subroutine sfclay_mynn
 
+!=================================================================================================================
+ end module module_sf_mynn
```
Comment (joeolson42) [inline src/core_atmosphere/physics/physics_wrf/module_sf_mynn.F]: I think this block should be removed. It appears to be duplicate code.

File: src/core_atmosphere/physics/physics_wrf/module_sf_mynn.F [hunk 31]
```diff
@@ -1,11 +1,17 @@
 !=================================================================================================================
  module module_sf_mynn
+ module module_sf_mynn
```
Comment (joeolson42) [inline src/core_atmosphere/physics/physics_wrf/module_sf_mynn.F]: duplicate line

File: src/core_atmosphere/physics/physics_wrf/module_sf_mynn.F [hunk 32]
```diff
@@ -1,11 +1,17 @@
...
 !=================================================================================================================
  use mpas_kind_types,only: RKIND,StrKIND
 
  use sf_mynn,only    : sf_mynn_run
  use sf_mynn_pre,only: sf_mynn_pre_run
 
+ use mpas_kind_types,only: RKIND,StrKIND
```
Comment (joeolson42) [inline src/core_atmosphere/physics/physics_wrf/module_sf_mynn.F]: duplicate line

Comment (joeolson42) [reply src/core_atmosphere/physics/physics_wrf/module_sf_mynn.F]: actually duplicate block

File: src/core_atmosphere/Registry.xml [hunk 33]
```diff
@@ -677,7 +701,7 @@
                         <var name="cldfrac_bl"/>
                         <var name="qc_bl"/>
                         <var name="qi_bl"/>
-			<var name="maxmf"/>
+			         <var name="maxmf"/>
```
Comment (joeolson42) [inline src/core_atmosphere/Registry.xml]: indentation  error in the merge

File: src/core_atmosphere/Registry.xml [hunk 34]
```diff
@@ -1607,34 +1704,35 @@
 
                         <var name="qc" array_group="moist" units="kg kg^{-1}"
                              description="Cloud water mixing ratio"
-                             packages="bl_mynn_in;bl_ysu_in;bl_myj_in;cu_ntiedtke_in;mp_kessler_in;mp_thompson_in;mp_wsm6_in;mp_nssl2m_in"/>
+                             packages="bl_mynn_in;bl_ysu_in;bl_myj_in;cu_nntiedtke_in;mp_kessler_in;mp_thompson_in;mp_thompson_aers_in;mp_wsm6_in;mp_nssl2m_in"/>
```
Comment (joeolson42) [inline src/core_atmosphere/Registry.xml]: "cu_nntiedtke_in" should be "cu_ntiedtke_in"

File: src/core_atmosphere/Registry.xml [hunk 35]
```diff
@@ -1607,34 +1704,35 @@
...
 
                         <var name="qr" array_group="moist" units="kg kg^{-1}"
                              description="Rain water mixing ratio"
-                             packages="mp_kessler_in;mp_thompson_in;mp_wsm6_in;mp_nssl2m_in"/>
+                             packages="mp_kessler_in;mp_thompson_in;mp_thompson_aers_in;mp_wsm6_in;mp_nssl2m_in"/>
 
                         <var name="qi" array_group="moist" units="kg kg^{-1}"
                              description="Ice mixing ratio"
-                             packages="bl_mynn_in;bl_ysu_in;bl_myj_in;cu_ntiedtke_in;mp_thompson_in;mp_wsm6_in;mp_nssl2m_in"/>
+                             packages="bl_mynn_in;bl_ysu_in;bl_myj_in;cu_nntiedtke_in;mp_thompson_in;mp_thompson_aers_in;mp_wsm6_in;mp_nssl2m_in"/>
```
Comment (joeolson42) [inline src/core_atmosphere/Registry.xml]: "cu_nntiedtke_in" should be "cu_ntiedtke_in"

File: src/core_atmosphere/Registry.xml [hunk 36]
```diff
@@ -1607,34 +1704,35 @@
...
 
                         <var name="qs" array_group="moist" units="kg kg^{-1}"
                              description="Snow mixing ratio"
-                             packages="bl_mynn_in;mp_thompson_in;mp_wsm6_in;mp_nssl2m_in"/>
+                             packages="bl_mynn_in;bl_mynn_in;mp_thompson_in;mp_thompson_aers_in;mp_wsm6_in;mp_nssl2m_in"/>
```
Comment (joeolson42) [inline src/core_atmosphere/Registry.xml]: duplicate bl_mynn_in

File: src/core_atmosphere/Registry.xml [hunk 37]
```diff
@@ -1983,45 +2105,46 @@
 
                         <var name="tend_qc" name_in_code="qc" array_group="moist" units="kg m^{-3} s^{-1}"
                              description="Tendency of cloud water mass per unit volume divided by d(zeta)/dz"
-                             packages="bl_mynn_in;bl_ysu_in;bl_myj_in;cu_ntiedtke_in;mp_kessler_in;mp_thompson_in;mp_wsm6_in;mp_nssl2m_in"/>
+                             packages="bl_mynn_in;bl_ysu_in;bl_myj_in;cu_nntiedtke_in;mp_kessler_in;mp_thompson_in;mp_thompson_aers_in;mp_wsm6_in;mp_nssl2m_in"/>
```
Comment (joeolson42) [inline src/core_atmosphere/Registry.xml]: This "cu_nntiedtke_in" error is happening so frequently, I'm starting to wonder if they actually changed the name of the scheme... Need to find out and do a regressive search and replace if necessary.

File: src/core_atmosphere/Registry.xml [hunk 38]
```diff
@@ -1983,45 +2105,46 @@
...
 
                         <var name="tend_qi" name_in_code="qi" array_group="moist" units="kg m^{-3} s^{-1}"
                              description="Tendency of ice mass per unit volume divided by d(zeta)/dz"
-                             packages="bl_mynn_in;bl_ysu_in;bl_myj_in;cu_ntiedtke_in;mp_thompson_in;mp_wsm6_in;mp_nssl2m_in"/>
+                             packages="bl_mynn_in;bl_ysu_in;bl_myj_in;cu_nntiedtke_in;mp_thompson_in;mp_thompson_aers_in;mp_wsm6_in;mp_nssl2m_in"/>
 
                         <var name="tend_qs" name_in_code="qs" array_group="moist" units="kg m^{-3} s^{-1}"
                              description="Tendency of snow mass per unit volume divided by d(zeta)/dz"
-                             packages="bl_mynn_in;mp_thompson_in;mp_wsm6_in;mp_nssl2m_in"/>
+                             packages="bl_mynn_in;bl_mynn_in;mp_thompson_in;mp_thompson_aers_in;mp_wsm6_in;mp_nssl2m_in"/>
```
Comment (joeolson42) [inline src/core_atmosphere/Registry.xml]: duplicate bl_mynn_in

File: src/core_atmosphere/physics/mpas_atmphys_driver_gwdo.F [hunk 39]
```diff
@@ -612,6 +612,7 @@ subroutine driver_gwdo(itimestep,configs,mesh,sfc_input,diag_physics,tend_physic
  gwdo_select: select case (trim(gwdo_scheme))
 
     case("bl_ysu_gwdo")
+       call mpas_timer_start('bl_gwdo')
```
Comment (joeolson42) [inline src/core_atmosphere/physics/mpas_atmphys_driver_gwdo.F]: duplicate line

File: src/core_atmosphere/CMakeLists.txt
Comment (joeolson42) [inline src/core_atmosphere/CMakeLists.txt:1]: Note to everyone: this new file needs to be modified to compile our schemes, at least MYNN.

**Review by joeolson42 — CHANGES_REQUESTED** [review]
I think I've done all I can with my own eyes. I'll have to wait until some other conflicts are resolved, then we can start compiling & debugging.

File: src/core_atmosphere/Registry.xml (hunk 37 above)
Comment (jderrico-noaa) [reply src/core_atmosphere/Registry.xml]: They didn't change the name of the scheme, I think this is a result of auto-combining some of the conflicted lines

**Comment by AndersJensen-NOAA** [comment]
> Merges v 8.2 into gsl/develop
> 
> There are a number of conflicts present that begin with "CURRENT GSL/DEVELOP VERSION" that need to be manually resolved before the model can actually be compiled/tested

I think I found all of the conflicts. @joeolson42 I ran into a compilation error: 
Screenshot 2024-08-22 at 8 36 36 PM

**Comment by tanyasmirnova** [comment]
> > Merges v 8.2 into gsl/develop
> > There are a number of conflicts present that begin with "CURRENT GSL/DEVELOP VERSION" that need to be manually resolved before the model can actually be compiled/tested
> 
> I think I found all of the conflicts. @joeolson42 I ran into a compilation error: Screenshot 2024-08-22 at 8 36 36 PM

@AndersJensen-NOAA Anders, I added this variable to the MYNN surface layer scheme because it is needed for 2-m diagnostics and CQS is not equal to CHS.

**Pull request merged**